### ISSUE TITLE
Setting 'required' option to false on a sonata_type_collection causes all nested form fields to lose their 'required' attributes

### ISSUE THREAD

**Issue by eikes** [sonata-project/SonataAdminBundle#2158]
I am not sure if this is a bug in the code or if it is sufficient to simply add a hint in the documentation. The problem occurs when I create a form field like this:

``` php
class CityAdmin extends Admin
{
    protected function configureFormFields(FormMapper $formMapper)
    {
        $formMapper
            ->add('name')
            ->add('districts', 
                'sonata_type_collection',
                array(
                    'by_reference' => false,
                    // I am talking about the next line
                    'required' => false, 
                    'type_options' => array(
                        'delete' => false,
                        'required' => true,
                    )
                ), 
                array(
                    'edit' => 'inline',
                    'inline' => 'table',
                )
            );
    }
}
```

Now the nested Admin class looks like this:

``` php
class DistrictAdmin extends Admin
{
    protected function configureFormFields(FormMapper $formMapper)
    {
        $formMapper
            ->add(
                'name',
                'text',
                array(
                    // this is implicit, but here for clarity
                    'required' => true
                )
            )
        ;
    }
}
```

When rows are added to the CityAdmin district field, the name field of the district is not required anymore even though it should be.

So setting the 'required' option in the sonata_type_collection to false has a nasty side effect. 

Now you might wonder why I would set this option to false. I did so to indicate to the user that it is not a requirement to add Districts to the City but simply optional (no \* after the label). What I didn't know is that setting 'required' to true doesn't actually cause a requirement of having at least one District, in stead the form still validates even if there are no Districts provided. So basically this is the desired behavior. 

The bottom line is, that setting the 'required' option does not have the effect I would have expected if set to true AND it has a nasty side effect if set to false. I think this should at least be mentioned in the docs. I can write a little _Tip_ and open a pull request if you want.

**Comment by core23** [comment]
This issue is very old. Can you please check this against the latest master version? 

Feel free to close if you don't use this bundle anymore or the issue is fixed. @eikes

**Comment by jordisala1991** [comment]
There was a doc PR to indicate this special behavior, and it was already merged.

Closing for now. If someone wants to provide. RFC issue on how this could be changed, feel free.
